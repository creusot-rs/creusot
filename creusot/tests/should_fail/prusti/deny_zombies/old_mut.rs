#![deny(creusot::prusti_zombie)]
extern crate creusot_contracts;
use creusot_contracts::prusti_prelude::*;

#[ensures(*x.0 == *result)]
pub fn zombie_old<'a, T>(x: (&'a mut T, u32)) -> &'a mut T {
    x.0
}