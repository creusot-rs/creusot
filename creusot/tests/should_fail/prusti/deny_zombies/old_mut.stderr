error: `(&'a mut T, u32)` cannot be moved from `'pre` to `'post` without becoming a zombie `(creusot_contracts::prusti::Zombie<&'a mut T>, u32)`
 --> old_mut.rs:5:12
  |
5 | #[ensures(*x.0 == *result)]
  |            ^
  |
note: the lint level is defined here
 --> old_mut.rs:1:9
  |
1 | #![deny(creusot::prusti_zombie)]
  |         ^^^^^^^^^^^^^^^^^^^^^^

error[creusot]: invalid mut dereference of zombie expression with lifetime `'a` in state `'post`
 --> old_mut.rs:5:12
  |
5 | #[ensures(*x.0 == *result)]
  |            ^^^

error: aborting due to 2 previous errors

